DataFormats/Provenance/src/ProductRegistry.cc

```diff
@@ -457,6 +458,51 @@ namespace edm {
     }
     checkDictionariesOfConsumedTypes(
         productTypesConsumed, elementTypesConsumed, containedTypeMap, containedTypeToBaseTypesMap);
+
```

Comment:
For clarity, would it be better to move all of this into a separate function?

Reply:
Thanks, will do.